app/k9mail/src/main/java/com/fsck/k9/account/AccountLoader.kt

```diff
@@ -0,0 +1,58 @@
+package com.fsck.k9.account
+
+import app.k9mail.feature.account.common.AccountCommonExternalContract
+import app.k9mail.feature.account.common.domain.entity.Account
+import app.k9mail.feature.account.common.domain.entity.AccountOptions
+import com.fsck.k9.Preferences
+import com.fsck.k9.logging.Timber
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import com.fsck.k9.Account as K9Account
+
+class AccountLoader(
+    private val preferences: Preferences,
+    private val coroutineDispatcher: CoroutineDispatcher = Dispatchers.IO,
+) : AccountCommonExternalContract.AccountLoader {
+
+    @Suppress("TooGenericExceptionCaught")
+    override suspend fun loadAccount(accountUuid: String): Account? {
+        return try {
+            withContext(coroutineDispatcher) {
+                load(accountUuid)
+            }
+        } catch (e: Exception) {
+            Timber.e(e, "Error while loading account")
+
+            null
+        }
+    }
+
+    private fun load(accountUuid: String): Account? {
+        val existingAccount = preferences.getAccount(accountUuid)
+
+        return if (existingAccount != null) {
+            mapToAccount(existingAccount)
+        } else {
+            null
+        }
```

Comment:
```suggestion
        return preferences.getAccount(accountUuid)?.let { mapToAccount(it) }
```

or with `mapToAccount()` converted to an extension function:

```suggestion
        return preferences.getAccount(accountUuid)?.mapToAccount()
```